bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetManager.java

```diff
@@ -506,8 +512,21 @@ public void startup(IProgressMonitor monitor) {
 		job = new CharsetManagerJob();
 		resourceChangeListener = new ResourceChangeListener();
 		workspace.addResourceChangeListener(resourceChangeListener, IResourceChangeEvent.POST_CHANGE);
+		initPreferenceChangeListener();
```

Comment:
Any particular reason why the listener is added before the job that is supposed to create markers? I would move this at very end of the method, just to keep them together.

Reply:
moved down to the end
(I've semantically-misgrouped the pref change listener w/ the resource change listener - but actually, they apparently form no semantic group)